### PR TITLE
feat(entity-registry): Enforce Field Path V2 uniqueness check during ingestion

### PR DESCRIPTION
**Pull request by pedro93**
Adds check in GMS when processing `schemaMetadata` aspects that all field paths are unique. 
If that is not the case, reject the MCP. 
The reason for this validator is that the UI will not be able to render schemas correctly if _field path ids_ are not unique.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: entity-registry/src/main/java/com/linkedin/metadata/aspect/validation/FieldPathValidator.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * Validates the Schema Field Path specification, specifically that all field IDs must be unique
+ * across all fields within a schema.
+ *
+ * @see <a href="https://datahubproject.io/docs/advanced/field-path-spec-v2/#requirements">Field
+ *     Path V2 docs</a>
+ */
+@Setter
+@Getter
+@Accessors(chain = true)
+public class FieldPathValidator extends AspectPayloadValidator {
```
Comment (david-leifker) [inline entity-registry/src/main/java/com/linkedin/metadata/aspect/validation/FieldPathValidator.java:32]: This needs an entry in the entity-registry to be activated.

**Comment by pedro93** [comment]
Failing CI seems unrelated.

**Review by david-leifker — CHANGES_REQUESTED** [review]
Handle patch or drop it as a supported operation please.

**Comment by jjoyce0510** [comment]
@david-leifker Let's get this one followed up on pls. Cheers!

**Comment by david-leifker** [comment]
There was some discussion around whether it was better to throw and exception here or to drop the duplicate field. I think the conclusion was to drop the duplicate field instead. @pedro93 - please let me know if I've mis-remember the conversation around this validator or not. If not, I'd close this PR.

**Comment by pedro93** [comment]
> There was some discussion around whether it was better to throw and exception here or to drop the duplicate field. I think the conclusion was to drop the duplicate field instead. @pedro93 - please let me know if I've mis-remember the conversation around this validator or not. If not, I'd close this PR.

I vaguely recall us wanting to throw an exception in the future **after** our connectors have been fixed. Silently dropping the duplicate fields was meant as a stop gap measure. @hsheth2 please confirm.

**Comment by hsheth2** [comment]
Based on our telemetry, about ~0.1% of ingestion runs produce duplicate field paths

Among the ingestions that needed to drop at least one field, it's a fairly small fraction of assets are actually impacted 
image

Given that this is a fairly small problem for ingestion and that ingestion has protections in place now, we're clear to include the server-side validation in the 0.13.4 release.

**Comment by david-leifker** [comment]
Replaced by https://github.com/datahub-project/datahub/pull/11619

**Pull request closed without merging**